Library/ViewModels/ProjectPamphletViewModel.swift

```diff
@@ -224,3 +192,29 @@ private func cookieFrom(refTag: RefTag, project: Project) -> HTTPCookie? {
 
   return HTTPCookie(properties: properties)
 }
+
+private func fetchProjectAndLiveStreams(projectOrParam: Either<Project, Param>)
+  -> SignalProducer<(Project, [LiveStreamEvent]?), NoError> {
+
+    let param = projectOrParam.ifLeft({ Param.id($0.id) }, ifRight: id)
+
+    let projectAndLiveStreams = AppEnvironment.current.apiService.fetchProject(param: param)
+      .ksr_delay(AppEnvironment.current.apiDelayInterval, on: AppEnvironment.current.scheduler)
+      .demoteErrors()
+      .flatMap { project -> SignalProducer<(Project, [LiveStreamEvent]?), NoError> in
+
+        AppEnvironment.current.liveStreamService
+          .fetchEvents(forProjectId: project.id, uid: AppEnvironment.current.currentUser?.id)
+          .ksr_delay(AppEnvironment.current.apiDelayInterval, on: AppEnvironment.current.scheduler)
+          .flatMapError { _ in SignalProducer(error: SomeError()) }
+          .timeout(after: 5, raising: SomeError(), on: AppEnvironment.current.scheduler)
+          .materialize()
+          .map { (project, .some($0.value?.liveStreamEvents ?? [])) }
```

Comment:
super subtle and glad tests helped make my intent clear! gina described it perfectly.